src/main/java/graphql/execution/ValuesResolver.java

```diff
@@ -9,11 +9,13 @@
 
 public class ValuesResolver {
 
-
     public Map<String, Object> getVariableValues(GraphQLSchema schema, List<VariableDefinition> variableDefinitions, Map<String, Object> inputs) {
         Map<String, Object> result = new LinkedHashMap<String, Object>();
         for (VariableDefinition variableDefinition : variableDefinitions) {
-            result.put(variableDefinition.getName(), getVariableValue(schema, variableDefinition, inputs.get(variableDefinition.getName())));
+            Object variableValue = getVariableValue(schema, variableDefinition, inputs.get(variableDefinition.getName()));
+            if (variableValue != null) {
+                result.put(variableDefinition.getName(), variableValue);
+            }
```

Comment:
Please disregard my previous comment, I was confused. 

But I am still confused. The `input` parameter may or may not be the result of deserialization, but at this stage of execution it doesn't matter where that `Map` came from. I'm sorry I'm not seeing this at all. A concrete example—or even better a failing test—would help a lot.